### PR TITLE
chore(deps): upgrade to .net 8 sdk

### PR DESCRIPTION
**Pull request by JamieMagee**
The packages are still published as targeting .NET 6, but this change requires .NET 8 to build the packages. This allows us to take advantage _some_ newer .NET features in a backwards compatible way.

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <Project>
 
   <PropertyGroup Label="Build">
-    <LangVersion>latest</LangVersion>
+    <LangVersion>preview</LangVersion>
```
Comment (kfcampbell) [inline Directory.Build.props:4]: I believe .NET 8 was officially released today. Would it be possible to switch to latest here?

File: samples/AspNetCore/MyWebhookEventProcessor.cs [hunk 2]
```diff
@@ -6,22 +6,18 @@
 using Octokit.Webhooks.Events;
 using Octokit.Webhooks.Events.PullRequest;
 
-public class MyWebhookEventProcessor : WebhookEventProcessor
+public class MyWebhookEventProcessor(ILogger<MyWebhookEventProcessor> logger) : WebhookEventProcessor
```
Comment (kfcampbell) [inline samples/AspNetCore/MyWebhookEventProcessor.cs:9]: Are these changes related to the update?

Comment (JamieMagee) [reply samples/AspNetCore/MyWebhookEventProcessor.cs:9]: Yeah, it's a new feature called [primary constructors][1]. Because we enable all warnings, and warnings as errors, the new SDK brings in [IDE0290][2] which forces us to adopt it.

It doesn't impact downstream consumers, as this is still compiled to the same CIL.

[1]: https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/tutorials/primary-constructors
[2]: https://learn.microsoft.com/en-gb/dotnet/fundamentals/code-analysis/style-rules/ide0290

File: Directory.Build.props (hunk 1 above)
Comment (JamieMagee) [reply Directory.Build.props:4]: I'm not sure if it was a bug in my setup, but in order to use primary constructors I had to set this to `preview` instead of `latest`.

We still target the .NET 6 runtime, but are using the .NET 8 SDK to compile our package. I wonder if it's just an issue with my setup, or if primary constructors in packages targeting .NET 6 require this to be `preview`.

Comment (JamieMagee) [reply Directory.Build.props]: It was an issue with my local setup. I tested with `latest` in a fresh codespace and it worked no problem.

**Review by kfcampbell — APPROVED** [review]
Thank you!
